packages/lodestar/src/tasks/index.ts

```diff
@@ -39,14 +42,30 @@ export class TasksService implements IService {
 
   public async start(): Promise<void> {
     this.chain.on("finalizedCheckpoint", this.handleFinalizedCheckpointChores);
+    this.chain.on("newEpoch", this.blockProcessingQueueCleanUp);
   }
 
   public async stop(): Promise<void> {
     this.chain.removeListener("finalizedCheckpoint", this.handleFinalizedCheckpointChores);
+    this.chain.removeListener("newEpoch", this.blockProcessingQueueCleanUp);
   }
 
   private handleFinalizedCheckpointChores = async (finalizedCheckpoint: Checkpoint): Promise<void> => {
     new ArchiveBlocksTask(this.config, {db: this.db, logger: this.logger}, finalizedCheckpoint).run();
   };
 
+  private blockProcessingQueueCleanUp = 
+  async (newEpoch: Epoch, blockProcessingQueue: FastPriorityQueue<IBlockProcessJob>): Promise<void> => {
+    return new Promise((resolve): void => {
+      while (!blockProcessingQueue.isEmpty()) {
+        const blockProcessJob = blockProcessingQueue.poll();
```

Comment:
poll method removes elements from queue which would also remove future blocks.
I suggest checking out `removeMany` method of https://www.npmjs.com/package/fastpriorityqueue